src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs

```diff
@@ -87,6 +87,10 @@ public Task<Stream> UpgradeAsync()
     {
         if (!IsUpgradableRequest)
         {
+            if (Request.ProtocolVersion < System.Net.HttpVersion.Version11)
+            {
+                throw new InvalidOperationException("Upgrade is not valid on connections targeting HTTP/1.0 or lower.");
+            }
             throw new InvalidOperationException("This request cannot be upgraded, it is incompatible.");
```

Comment:
Does this sound reasonable?

```suggestion
            throw new InvalidOperationException("This request cannot be upgraded, because it has a body.");
```